src/agent/onefuzz-agent/src/local/coverage.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+use crate::{
+    local::common::{
+        build_local_context, get_cmd_arg, get_cmd_env, get_cmd_exe, get_synced_dir,
+        get_synced_dirs, CmdType, CHECK_FUZZER_HELP, COVERAGE_DIR, COVERAGE_FILTER, INPUTS_DIR,
+        READONLY_INPUTS, TARGET_ENV, TARGET_EXE, TARGET_OPTIONS, TARGET_TIMEOUT,
+    },
+    tasks::{
+        config::CommonConfig,
+        coverage::generic::{Config, CoverageTask},
+    },
+};
+use anyhow::Result;
+use clap::{App, Arg, SubCommand};
+use flume::Sender;
+use storage_queue::QueueClient;
+
+use super::common::{SyncCountDirMonitor, UiEvent};
+
+pub fn build_coverage_config(
+    args: &clap::ArgMatches<'_>,
+    local_job: bool,
+    input_queue: Option<QueueClient>,
+    common: CommonConfig,
+    event_sender: Option<Sender<UiEvent>>,
+) -> Result<Config> {
+    let target_exe = get_cmd_exe(CmdType::Target, args)?.into();
+    let target_env = get_cmd_env(CmdType::Target, args)?;
+    let target_options = get_cmd_arg(CmdType::Target, args);
+    let target_timeout = value_t!(args, TARGET_TIMEOUT, u64).ok();
+    let coverage_filter = value_t!(args, TARGET_TIMEOUT, String).ok();
+
+    let readonly_inputs = if local_job {
+        vec![
+            get_synced_dir(INPUTS_DIR, common.job_id, common.task_id, args)?
+                .monitor_count(&event_sender)?,
+        ]
+    } else {
+        get_synced_dirs(READONLY_INPUTS, common.job_id, common.task_id, args)?
```

Comment:
Double-checking: we really don't want to also include `INPUTS_DIR` in this branch?

Reply:
This mirrors the libfuzzer-coverage directly.

Note, the arg builder makes it such that INPUTS and READONLY_INPUTS are used differently depending on the caller.

Reply:
See the block of code from this PR I commented on with "(highlighting for other comment)".